CommonControls/PackFileBrowser/ContextMenuHandler.cs

```diff
@@ -127,11 +136,61 @@ void OnAddFilesCommand()
                 foreach (var file in files)
                 {
                     var fileName = Path.GetFileName(file);
+
                     var packFile = new PackFile(fileName, new MemorySource(File.ReadAllBytes(file)));
+
                     _packFileService.AddFileToPack(_selectedNode.FileOwner, parentPath, packFile);
                 }
             }
         }
+        void OnAddFilesWithConversionCommand()
+        {
+            var parentPath = _selectedNode.GetFullPath(); // get pack path, at mouse pointer        
```

Comment:
This logic does not belong here. It should be in a different class. 
This layer should as much as possible try to only route information.